src/Fixer/PhpTag/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,242 @@
+    {
+        $sample = <<<'EOT'
+<?=1?>
+<?php print '2' . '3'; ?>
+<?php /* comment */ echo '2' . '3'; ?>
+<?php print '2' . '3'; someFunction(); ?>
+
+EOT
+        ;
+
+        return new FixerDefinition(
+            'Replace short-echo `<?=` with long format `<?php echo`/`<?php print` syntax, or vice-versa.',
+            [
+                new CodeSample($sample),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG, self::OPTION_LONG_FUNCTION => self::LONG_FUNCTION_PRINT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT, self::OPTION_SHORT_ALWAYS => true]),
+            ],
+            null
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            if (null !== $tokens->findSequence([[\T_OPEN_TAG], [\T_ECHO]])) {
+                return true;
+            }
+            if (null !== $tokens->findSequence([[\T_OPEN_TAG], [\T_PRINT]])) {
+                return true;
+            }
+
+            return false;
+        }
+
+        return $tokens->isTokenKindFound(\T_OPEN_TAG_WITH_ECHO);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(self::OPTION_FORMAT, 'How the fixer should process short/long echo tags'))
+                ->setAllowedValues($this->supportedFormatOptions)
+                ->setDefault(self::FORMAT_LONG)
+                ->getOption(),
+            (new FixerOptionBuilder(self::OPTION_LONG_FUNCTION, 'The function to be used to expand the short echo tags'))
+                ->setAllowedValues($this->supportedLongFunctionOptions)
+                ->setDefault(self::LONG_FUNCTION_ECHO)
+                ->getOption(),
+            (new FixerOptionBuilder(self::OPTION_SHORT_ALWAYS, 'Always render short-echo tags even in case of complex code'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(false)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            $this->longToShort($file, $tokens);
+        } else {
+            $this->shortToLong($file, $tokens);
+        }
+    }
+
+    private function longToShort(SplFileInfo $file, Tokens $tokens)
+    {
+        $offset = 0;
+        for (;;) {
+            $foundEcho = $tokens->findSequence([[\T_OPEN_TAG], [\T_ECHO]], $offset);
+            $foundPrint = $tokens->findSequence([[\T_OPEN_TAG], [\T_PRINT]], $offset);
```

Comment:
Searching bor both and lower using only one - seems like waste of CPU.
Why not with classic `for` loop:
```
for ($i = 0 ; $i < count; $i++)
    if $i is not T_OPEN_TAG -> continue;
    $nextMeanigful = ...;
    if $nextMeanigful is not echo or print -> continue;
    if for complex option
    override
```

Reply:
Done: 5b3d5eb6ce6373aa1cfdf73752814061f7913ed7